src/s1ap/handlers/s1setup.c

```diff
@@ -133,13 +136,50 @@ create_s1setup_response(/*enb info,*/unsigned char **s1_setup_resp)
 	return resp_buf.pos;
 }
 
+/**
+ * @brief get socket info and add assoc info to hash
+ * @return error code.
+ */
+int
+addEnbInfoToHash(int enbfd, int inStreamId)
+{
+    struct sctp_status stat = {0};
+    struct enb_assoc_info enbAssoc = {0};
+    int stat_len = sizeof(struct sctp_status);
+    memset (&stat, 0, sizeof (struct sctp_status));
+    memset (&enbAssoc, 0, sizeof (struct enb_assoc_info));
+					
+    if (getsockopt (enbfd, IPPROTO_SCTP, 
+                  SCTP_STATUS, &stat, (socklen_t *)&stat_len) < 0) 
+    {
+	log_msg(LOG_ERROR, "Getsockopt failed: %s\n", 
+	                             strerror (errno));
+	return E_FAIL;
+    }
+
+    enbAssoc.enb_fd = enbfd;
+    enbAssoc.inStreamId = inStreamId;
+    enbAssoc.assoc_max_out_streams = stat.sstat_outstrms; 
+    enbAssoc.assoc_max_in_streams  = stat.sstat_instrms; 
+    enbAssoc.assocId = stat.sstat_assoc_id;
+
+    return onf_ht_insert(g_enbfd_hash_table, sizeof(int),
+       sizeof(struct enb_assoc_info), (const char*)&enbfd,
+       (const char*)&enbAssoc);
```

Comment:
It appears like enbAssoc which is local stack variable is added into the hash. And hash keeps the pointer to this ?

Reply:
the onf_ht_insert calls the onf_ht_new_item which does a malloc for the key len and value len.